src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -265,7 +275,16 @@
     </IcallTableGenerator>
   </Target>
 
-  <Target Name="_WasmSelectRuntimeComponentsForLinking" Condition="'$(WasmNativeWorkload)' == true" DependsOnTargets="_MonoSelectRuntimeComponents" />
+  <Target Name="_WasmSelectRuntimeComponentsForLinkingWorkload" Condition="'$(WasmNativeWorkload)' == 'true'" DependsOnTargets="_MonoSelectRuntimeComponents" />
+  <Target Name="_WasmSelectRuntimeComponentsForLinkingInTree" Condition="'$(WasmNativeWorkload)' != 'true'" >
+    <ItemGroup Condition="'$(WasmBuildNative)' == 'true'">
+      <_MonoRuntimeComponentDontLink Include="libmono-component-diagnostics_tracing-static.a"/>
+    </ItemGroup>
+    <ItemGroup Condition="'$(WasmBuildNative)' == 'true' and '$(Configuration)' != 'Debug'">
```

Comment:
We should handle the other case here as well, to avoid linking with both non-stub and stub components.